lib/tapioca/dsl/compilers/url_helpers.rb

```diff
@@ -134,17 +143,21 @@ def gather_non_discoverable_includers
             end.freeze
           end
 
+          # Returns `true` if `mod` "directly" includes `helper`.
+          # For classes, this method will return false if the `helper` is included only by a superclass
           sig { params(mod: Module, helper: Module).returns(T::Boolean) }
           private def includes_helper?(mod, helper)
-            superclass_ancestors = []
+            ancestors = ancestors_of(mod)
 
-            if Class === mod
-              superclass = superclass_of(mod)
-              superclass_ancestors = ancestors_of(superclass) if superclass
+            unique_ancestors = if Class === mod && (superclass = superclass_of(mod))
```

Comment:
This list isn't necessarily `unique_ancestors`, it is more like `own_ancestors` or something.